src/main/resources/static/js/subscription.js

```diff
@@ -809,28 +843,93 @@ jQuery(document).ready(function () {
             var regExpression = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
             if (!regExpression.test(notificationMeta) && notificationMeta != "") {
                 $('#invalidNotificationMeta').text("Not a valid email.");
-                $('#notificationMeta').addClass("is-invalid");
                 error = true;
             }
         }
         if (error) {
+            $('#notificationMeta').addClass("is-invalid");
             $('#invalidNotificationMeta').show();
         }
         return error;
     }
 
+    function validateMessageRawJson(jsonData, allowEmpty) {
+        var error = false;
+        $('#notificationMessageKeyError').hide();
+        $('#notificationMessageValuesError').hide();
+        $('#notificationMessageValuesJsonError').hide();
+        $('#invalidNotificationMessageRawJson').hide();
+        $('#notificationMessageRawJson').removeClass("is-invalid");
+
+        if (!allowEmpty && jsonData == "") {
+            $('#invalidNotificationMessageRawJson').text("The raw body field must not be empty.");
+            error = true;
+        }
+
+        if (jsonData != "" && !validateJsonString(jsonData)) {
+            $('#invalidNotificationMessageRawJson').text("Input is not valid JSON string.");
```

Comment:
This is a simplified JSON validator that converts a string into JSON, it may convert " " to JSON but it also validates [["":""], ["":""]] So if the user enters JSON as raw JSON the validation will fail if the json is wrong. The user do not need to enter string tags "". If the user does the JSON will not be validated. It would require a too complex implementation of a validator to validate more then that.

Reply:
Oh i see.  Then it should not say "Input is not valid JSON string since that would imply:
"{key:  value}"
instead of:
{key: value}

Reply:
Done.